data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/contingency/CsaProfileContingencyCreator.java

```diff
@@ -0,0 +1,151 @@
+            return;
+        }
+
+        if (!mustStudy) {
+            csaProfileContingencyCreationContexts.add(CsaProfileContingencyCreationContext.notImported(contingencyId, contingencyId, contingencyName, ImportStatus.INCONSISTENCY_IN_DATA, "contingency.mustStudy is false"));
+            return;
+        }
+
+        PropertyBags contingencyEquipments = CsaProfileCracUtils.getLinkedPropertyBags(contingencyEquipmentsPropertyBags, contingencyPropertyBag, CsaProfileConstants.REQUEST_CONTINGENCY, CsaProfileConstants.REQUEST_CONTINGENCY);
+        if (contingencyEquipments.isEmpty()) {
+            csaProfileContingencyCreationContexts.add(CsaProfileContingencyCreationContext.notImported(contingencyId, contingencyId, contingencyName, ImportStatus.INCONSISTENCY_IN_DATA, "no contingency equipment linked to the contingency"));
+            return;
+        }
+
+        boolean isIncorrectContingentStatus = false;
+        boolean isMissingNetworkElement = false;
+        String incorrectContingentStatusElements = "";
+        String missingNetworkElements = "";
+        for (PropertyBag contingencyEquipmentPropertyBag : contingencyEquipments) {
+            String equipmentId = contingencyEquipmentPropertyBag.getId(CsaProfileConstants.REQUEST_CONTINGENCIES_EQUIPMENT_ID);
+            String contingentStatus = contingencyEquipmentPropertyBag.get(CsaProfileConstants.REQUEST_CONTINGENCIES_CONTINGENT_STATUS);
+
+            if (!CsaProfileConstants.IMPORTED_CONTINGENT_STATUS.equals(contingentStatus)) {
+                isIncorrectContingentStatus = true;
+                incorrectContingentStatusElements = incorrectContingentStatusElements.concat(equipmentId + " ");
+            } else {
+                String networkElementId = this.getNetworkElementIdInNetwork(equipmentId);
+                if (networkElementId == null) {
+                    isMissingNetworkElement = true;
+                    missingNetworkElements = missingNetworkElements.concat(equipmentId + " ");
+                } else {
+                    contingencyAdder.withNetworkElement(networkElementId);
+                }
+            }
+        }
+
+        if (isIncorrectContingentStatus) {
+            csaProfileContingencyCreationContexts.add(CsaProfileContingencyCreationContext.notImported(contingencyId, contingencyId, contingencyName, ImportStatus.INCONSISTENCY_IN_DATA, "incorrect contingent status for equipments : " + incorrectContingentStatusElements));
+            return;
+        }
+
+        if (isMissingNetworkElement) {
+            csaProfileContingencyCreationContexts.add(CsaProfileContingencyCreationContext.notImported(contingencyId, contingencyId, contingencyName, ImportStatus.INCONSISTENCY_IN_DATA, "missing contingency equipments in network : " + missingNetworkElements));
+            return;
```

Comment:
in this case you should still import the contingency but activate the "altered" flag (since you've alreay done the work of selecting network elements that are valid). Us importStatusDetail to say which network elements were excluded because invalid. Unless all elements are invalid, in which case you should not import the contingency.

Reply:
done, if network elements are missing, and also if contigent status are incorrect for some elements